custom/src/test/java/co/elastic/otel/SpanStackTraceTest.java

```diff
@@ -45,81 +57,98 @@ void resetGlobalOtel() {
+        ElasticAutoConfigurationCustomizerProvider.STACKTRACE_LEGACY2_DURATION
+      })
+  void featureCanBeDisabled(String configName) {
     try (AutoConfigTestProperties testProps =
-        new AutoConfigTestProperties()
-            .put(SpanStackTraceProcessorAutoConfig.MIN_DURATION_CONFIG_OPTION, "-1")) {
+        new AutoConfigTestProperties().put(configName, "-1")) {
       OpenTelemetry otel = GlobalOpenTelemetry.get();
 
       assertThat(OtelReflectionUtils.getSpanProcessors(otel))
           .noneSatisfy(proc -> assertThat(proc).isInstanceOf(StackTraceSpanProcessor.class));
     }
   }
 
-  @Test
-  void legacyConfigOptionSupported() {
+  @ParameterizedTest
+  @ValueSource(
+      strings = {
+        ElasticAutoConfigurationCustomizerProvider.STACKTRACE_OTEL_DURATION,
+        ElasticAutoConfigurationCustomizerProvider.STACKTRACE_LEGACY1_DURATION,
+        ElasticAutoConfigurationCustomizerProvider.STACKTRACE_LEGACY2_DURATION
+      })
+  void legacyConfigOptionsSupported(String configName) {
     try (AutoConfigTestProperties testProps =
-        new AutoConfigTestProperties()
-            .put(SpanStackTraceProcessorAutoConfig.LEGACY_DURATION_CONFIG_OPTION, "0ms")) {
+        new AutoConfigTestProperties().put(configName, "0ms")) {
       OpenTelemetry otel = GlobalOpenTelemetry.get();
 
-      // TODO: cleanup other extensions (Breakdown) so that this is not required:
-      ElasticExtension.INSTANCE.registerOpenTelemetry(otel);
+      Tracer tracer = otel.getTracer("test-tracer");
+
+      tracer.spanBuilder("my-span").startSpan().end();
+
+      checkSpanStackTrace(true);
+    }
+  }
+
+  @Test
+  void checkMinDurationRespected() {
+    try (AutoConfigTestProperties testProps =
+        new AutoConfigTestProperties().put(OTEL_MIN_DURATION, "100s")) {
+      OpenTelemetry otel = GlobalOpenTelemetry.get();
 
       Tracer tracer = otel.getTracer("test-tracer");
 
       tracer.spanBuilder("my-span").startSpan().end();
 
-      List<SpanData> spans = AutoConfiguredDataCapture.getSpans();
+      checkSpanStackTrace(false);
+    }
+  }
 
-      assertThat(spans).hasSize(1);
-      assertThat(spans.get(0))
-          .hasName("my-span")
+  private void checkSpanStackTrace(boolean stackTraceExpected) {
+    List<SpanData> spans = AutoConfiguredDataCapture.getSpans();
+    assertThat(spans).hasSize(1);
+    SpanData span = spans.get(0);
+    if (stackTraceExpected) {
+      assertThat(span)
           .hasAttribute(
-              satisfies(CodeIncubatingAttributes.CODE_STACKTRACE, att -> att.isNotBlank()));
+              satisfies(
+                  CodeIncubatingAttributes.CODE_STACKTRACE,
+                  AbstractCharSequenceAssert::isNotBlank));
+    } else {
+      assertThat(span.getAttributes().get(CodeIncubatingAttributes.CODE_STACKTRACE)).isNull();
     }
   }
 
   @Test
-  void checkMinDurationRespected() {
+  void checkInferredSpansIgnored() {
```

Comment:
[for reviewer] this implicitly tests that the feature is working as expected, in particular that the filter configuration has been properly set and is working as expected.